### PR TITLE
Add org.gnome.Boxes.Extension.OsinfoDb

### PR DESCRIPTION
**Pull request by felipeborges**
This is a GNOME Boxes extension that will update the operating system's data that Boxes uses for setting up virtual machines.

### CODE REVIEW

**Comment by nedrichards** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

**Comment by nedrichards** [comment]
@felipeborges this looks like the upstream "master" version rather than something aimed at flathub which would need stable versions and pinned commits etc. you probably need a skip-appstream-checker flathub.json as well.

**Comment by felipeborges** [comment]
@nedrichards osinfo-db is really just a database with VMs preferences, we wanted to have it as up-to-date as possible, that's why I went with "master".

Oh, I can actually add appdata.

**Comment by felipeborges** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

**Comment by felipeborges** [comment]
Now the build fails because we are missing the extension data in org.gnome.Boxes.json.

Will I break the Boxes build if I add the extension metadata to its manifest before we merge this? It works locally.

Also, I updated the extension's manifest to refer to org.gnome.Boxes//stable and use org.gnome.Sdk//3.36. Still, I would like to pull osinfo-db from its master branch.

**Comment by nedrichards** [comment]
Yeah, the source of the bits that you put here can be master - but it should be a pinned commit at least. If this is going to be absolutely required then it might be better being exported out of the existing boxes repo - I'll have a quick think about what will work best to keep this "fresh" and reduce complexity.

**Comment by felipeborges** [comment]
Pinning a commit should be fine. My main goal here is to decouple the OS info updates from the Boxes updates.

Osinfo updates are small but nice to have, especially URL updates when distros go EOL, when vendors move ISOs around breaking our URLs, when vendors release .1 versions, etc...

Besides, we have Boxes stable, beta, unstable and we also plan on having Boxes Flatpaks build from fedora/rhel RPMs. All these versions could consume the osinfo updates from the extension.

**Comment by barthalion** [comment]
> Will I break the Boxes build if I add the extension metadata to its manifest before we merge this? It works locally.

No, there will be just nothing to satisfy that extension point, but it won't cause issues.

**Comment by felipeborges** [comment]
> if this is going to be absolutely required then it might be better being exported out of the existing boxes repo

@nedrichards this extension isn't necessarily required. Boxes will still ship its own osinfo-db, but with the extension we can deliver updates to it.

**Comment by felipeborges** [comment]
Any follow ups on this? I merged https://github.com/flathub/org.gnome.Boxes/pull/53 in Boxes.

Besides, we now have very strict CI gating in osinfo-db upstream.

**Comment by felipeborges** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

**Comment by barthalion** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

File: org.gnome.Boxes.Extension.OsinfoDb.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
+{
+    "id" : "org.gnome.Boxes.Extension.OsinfoDb",
+    "branch" : "master",
+    "runtime" : "org.gnome.Boxes",
+    "runtime-version" : "stable",
+    "sdk" : "org.gnome.Sdk//3.36",
```
Comment (barthalion) [inline org.gnome.Boxes.Extension.OsinfoDb.json:6]:
```suggestion
    "sdk" : "org.gnome.Sdk//3.38",
```

File: org.gnome.Boxes.Extension.OsinfoDb.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
+{
+    "id" : "org.gnome.Boxes.Extension.OsinfoDb",
+    "branch" : "master",
```
Comment (barthalion) [inline org.gnome.Boxes.Extension.OsinfoDb.json:3]:
```suggestion
    "branch" : "master",
```

File: org.gnome.Boxes.Extension.OsinfoDb.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            "buildsystem" : "simple",
+            "builddir": true,
+            "build-commands" : [
+                "make",
+                "osinfo-db-import --dir /app/share/osinfo/ osinfo-db-*.tar.xz"
+            ],
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://gitlab.com/libosinfo/osinfo-db.git"
```
Comment (barthalion) [inline org.gnome.Boxes.Extension.OsinfoDb.json:36]: It should use a specific commit as we want builds to be reproducible.

**Review by barthalion — COMMENTED** [review]
I'm not really sure if extension definition in Boxes is correct.

```
    "add-extensions" : {
        "org.gnome.Boxes.Extension" : {
            "version" : "master",
            "directory" : "share/osinfo",
            "no-autodownload" : false,
            "autodelete" : true
        }
    },
```

It should specify full extension name given it specifies `share/osinfo` for the mount point and also use `stable` for the version.

File: org.gnome.Boxes.Extension.OsinfoDb.json (hunk 2 above)
Comment (felipeborges) [reply org.gnome.Boxes.Extension.OsinfoDb.json:3]: sorry, should this be "stable" too? It is unclear to me whether this refer to the flatpak branching or the osinfo-db git branch name.

**Comment by felipeborges** [comment]
thanks a lot for your review!

File: org.gnome.Boxes.Extension.OsinfoDb.json (hunk 2 above)
Comment (barthalion) [reply org.gnome.Boxes.Extension.OsinfoDb.json:3]: This is flatpak branch, Flathub automation sets it automatically based on git branch. Usually there's no need to specify it.

**Comment by barthalion** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

**Comment by barthalion** [comment]
The last missing piece is `skip-icons-check` like here: https://github.com/flathub/org.freedesktop.Sdk.Extension.golang/blob/branch/19.08/flathub.json#L2

**Comment by felipeborges** [comment]
bot, build org.gnome.Boxes.Extension.OsinfoDb

**Comment by barthalion** [comment]
:tada: 

/merge

**Comment by felipeborges** [comment]
Thank you very much!

**Pull request closed without merging**